src/openrct2/Game.cpp

```diff
@@ -726,47 +718,39 @@ static void limit_autosave_count(const size_t numberOfFilesToKeep, bool processL
         return;
     }
 
-    autosaveFiles = static_cast<utf8**>(malloc(sizeof(utf8*) * autosavesCount));
-
+    auto autosaveFiles = std::vector<std::string>(autosavesCount);
     {
         auto scanner = std::unique_ptr<IFileScanner>(Path::ScanDirectory(filter, false));
         for (size_t i = 0; i < autosavesCount; i++)
         {
-            autosaveFiles[i] = static_cast<utf8*>(malloc(sizeof(utf8) * MAX_PATH));
-            std::memset(autosaveFiles[i], 0, sizeof(utf8) * MAX_PATH);
-
+            autosaveFiles[i].resize(MAX_PATH, 0);
```

Comment:
Not sure we need this, but I'm a bit sleepy, :X

Reply:
You will need to do this because the string is being used as an array of data. I assume in the future the idea would be to correct the platform functions to return a std::string